dotnet/samples/KernelSyntaxExamples/Example65_HandlebarsPlanner.cs

```diff
@@ -0,0 +1,216 @@
+
+            {{!-- Step 2: Generate the novel outline with the given chapter count --}}
+            {{set "novelOutline" (WriterPlugin-NovelOutline input=(get "bookTitle") chapterCount=(get "chapterCount"))}}
+
+            {{!-- Step 3: Loop through the chapters and generate the content for each chapter --}}
+            {{#each (range 1 (get "chapterCount"))}}
+                {{set "chapterIndex" this}}
+                {{set "chapterSynopsis" (MiscPlugin-ElementAtIndex input=(get "novelOutline") index=(get "chapterIndex"))}}
+                {{set "previousChapterSynopsis" (MiscPlugin-ElementAtIndex input=(get "novelOutline") index=(get "chapterIndex" - 1))}}
+                
+                {{!-- Step 4: Write the chapter content using the WriterPlugin-NovelChapter helper --}}
+                {{set "chapterContent" (WriterPlugin-NovelChapter input=(get "chapterSynopsis") theme=(get "bookTitle") previousChapter=(get "previousChapterSynopsis") chapterIndex=(get "chapterIndex"))}}
+                
+                {{!-- Step 5: Output the chapter content --}}
+                {{json (get "chapterContent")}}
+            {{/each}}
+        */
+    }
+
+    /// <summary>
+    /// Plugin example with two native functions, where one function gets a random word and the other returns a definition for a given word.
+    /// </summary>
+    private sealed class DictionaryPlugin
+    {
+        public const string PluginName = nameof(DictionaryPlugin);
+
+        private readonly Dictionary<string, string> _dictionary = new()
+        {
+            {"apple", "a round fruit with red, green, or yellow skin and a white flesh"},
+            {"book", "a set of printed or written pages bound together along one edge"},
+            {"cat", "a small furry animal with whiskers and a long tail that is often kept as a pet"},
+            {"dog", "a domesticated animal with four legs, a tail, and a keen sense of smell that is often used for hunting or companionship"},
+            {"elephant", "a large gray mammal with a long trunk, tusks, and ears that lives in Africa and Asia"}
+        };
+
+        [SKFunction, SKName("GetRandomWord"), System.ComponentModel.Description("Gets a random word from a dictionary of common words and their definitions.")]
+        public string GetRandomWord()
+        {
+            // Get random number
+            var index = RandomNumberGenerator.GetInt32(0, this._dictionary.Count - 1);
+
+            // Return the word at the random index
+            return this._dictionary.ElementAt(index).Key;
+        }
+
+        [SKFunction, SKName("GetDefinition"), System.ComponentModel.Description("Gets the definition for a given word.")]
+        public string GetDefinition([System.ComponentModel.Description("Word to get definition for.")] string word)
+        {
+            return this._dictionary.TryGetValue(word, out var definition)
+                ? definition
+                : "Word not found";
+        }
+    }
```

Comment:
nit: Can we have an example that shows a plugin that uses complex types? It's OK if this comes in a follow up PR.

Reply:
Yep will come in follow-up PR